aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -239,6 +241,128 @@ Tensor wrapped_scalar_tensor(
 
 } // anonymous namespace
 
+Tensor& isin_out(const Tensor& elements, const Tensor& test_elements, bool assume_unique, bool invert, Tensor& out) {
+  out.resize_as_(elements);
+  if (elements.numel() == 0) {
+    return out;
+  }
+
+  // Condition taken from numpy's implementation.
+  Tensor test_elements_flat = test_elements.ravel();
+  if (test_elements_flat.numel() < static_cast<int64_t>(
+        10.0f * std::pow(static_cast<double>(elements.numel()), 0.145))) {
+    auto iter = TensorIteratorConfig()
+      .set_check_mem_overlap(false)
+      .add_output(out)
+      .add_input(elements)
+      .cast_common_dtype_to_outputs(false)
+      .enforce_safe_casting_to_output(false)
+      .check_all_same_dtype(false)
+      .build();
+    isin_default_stub(iter.device_type(), iter, test_elements_flat, invert);
+    return out;
+  }
+
+  // If the number of test elements is large, use a sorting-based algorithm:
+  // 1. Concatenate unique elements with unique test elements in 1D form. If
+  //    assume_unique is true, skip calls to unique().
+  // 2. Stable sort all elements, maintaining order indices to reverse the
+  //    operation. Stable sort is necessary to keep elements before test
+  //    elements within the sorted list.
+  // 3. Create a mask for locations of adjacent duplicate values within the
+  //    sorted list. Duplicate values are in both elements and test elements.
+  // 4. Reorder the mask to match the pre-sorted element order.
+  // 5. Index the mask to match the pre-unique element order. If
+  //    assume_unique is true, just take the first N items of the mask,
+  //    where N is the original number of elements.
+
+  Tensor elements_flat, unique_order;
+  if (assume_unique) {
+    elements_flat = elements.ravel();
+  } else {
+    std::tie (elements_flat, unique_order) = at::_unique(
+        elements, /*sorted=*/ false, /*return_inverse=*/ true);
+    std::tie (test_elements_flat, std::ignore) = at::_unique(test_elements_flat, /*sorted=*/ false);
+  }
+
+  Tensor all_elements = at::_cat({elements_flat, test_elements_flat});
+  Tensor sorted_elements, sorted_order;
+  std::tie (sorted_elements, sorted_order) = all_elements.sort(
+      /*stable=*/ c10::optional<bool>(true), /*dim=*/ 0, /*descending=*/ false);
```

Comment:
Can we just use `true` instead of  `c10::optional<bool>(true)`, I believe it can be auto casted, correct me if I am wrong. Since our next step for stable sort is to make this optional bool to bool I guess.

Reply:
Yep it works! Fixed